src/Sylius/Behat/Context/Api/Admin/ManagingTaxRatesContext.php

```diff
@@ -0,0 +1,526 @@
+     */
+    public function iShouldBeNotifiedThatTaxRateWithThisCodeAlreadyExists(): void
+    {
+        $response = $this->client->getLastResponse();
+        Assert::false(
+            $this->responseChecker->isCreationSuccessful($response),
+            'Tax rate has been created successfully, but it should not'
+        );
+        Assert::same(
+            $this->responseChecker->getError($response),
+            'code: The tax rate with given code already exists.'
+        );
+    }
+
+    /**
+     * @Then there should still be only one tax rate with code :code
+     */
+    public function thereShouldStillBeOnlyOneTaxRateWithCode(string $code): void
+    {
+        Assert::count(
+            $this->responseChecker->getCollectionItemsWithValue($this->client->index(Resources::TAX_RATES), 'code', $code),
+            1,
+            sprintf('There is more than one tax rate with code %s', $code)
+        );
+    }
+
+    /**
+     * @Then I should be notified that :element is required
+     */
+    public function iShouldBeNotifiedThatCodeIsRequired(string $element): void
+    {
+        Assert::contains(
+            $this->responseChecker->getError($this->client->getLastResponse()),
+            sprintf('%s: Please enter tax rate %s.', $element, $element)
+        );
+    }
+
+    /**
+     * @Then tax rate with :element :code should not be added
+     */
+    public function taxRateWithCodeShouldNotBeAdded(string $element, string $code): void
+    {
+        Assert::false($this->isItemOnIndex($element, $code), sprintf('Tax rate with %s %s exist', $element, $code));
+    }
+
+    private function isItemOnIndex(string $property, string $value): bool
+    {
+        return $this->responseChecker->hasItemWithValue($this->client->index(Resources::TAX_RATES), $property, $value);
+    }
+
+    /**
+     * @Then I should be notified that zone has to be selected
+     */
+    public function iShouldBeNotifiedThatZoneHasToBeSelected(): void
+    {
+        Assert::contains(
+            $this->responseChecker->getError($this->client->getLastResponse()),
+            'zone: Please select tax zone.'
+        );
+    }
+
+    /**
+     * @Then I should be notified that category has to be selected
+     */
+    public function iShouldBeNotifiedThatCategoryHasToBeSelected(): void
+    {
+        Assert::contains(
+            $this->responseChecker->getError($this->client->getLastResponse()),
+            'category: Please select tax category.'
+        );
+    }
+
+    /**
+     * @Then I should not see a tax rate with name :name
+     */
+    public function iShouldNotSeeATaxRateWithName(string $name): void
+    {
+        Assert::false(
+            $this->responseChecker->hasItemWithValue($this->client->getLastResponse(), 'name', $name),
+            sprintf('Tax rate with name %s exists', $name)
+        );
+    }
+
+    /**
+     * @When /^I want to modify (this tax rate)$/
```

Comment:
It would be nice to maintain the order of methods in contexts: Given -> When -> Then